packages/compiler/core/checker.ts

```diff
@@ -1621,6 +1690,89 @@ export function createChecker(program: Program): Checker {
     properties.set(newProp.name, newProp);
   }
 
+  /**
+   * Initializes a late bound symbol for the type. This is generally necessary when attempting to
+   * access a symbol for a type that is created during the check phase.
+   */
+  function lateBindMemberContainer(type: ModelType | InterfaceType | UnionType) {
```

Comment:
Just to make sure I understand. Cases on when we need late binding:
- template parameters are not instantiated yet in binder so they can't have symbol
- each members of model, enums etc might not have the complete picture(spread, is, extends) until it is fully resolved in the checker

Reply:
Template parameters do get a symbol in locals, but that is kind of unrelated since you can't dot into those symbols.

The second part is correct. The other part is that, if a template type is instantiated in the checker, a symbol for the type itself is created. I won't claim that late binding is "needed" here, it's just how I did it but I'm not super happy with it.

Reply:
sorry, that's what I meant for the 1st point, not sure why I said template parameters instead of templates